rust/schema.py

```diff
@@ -31,30 +33,514 @@ class Location(Element):
+#         label: Option<LabelId>,
+#     },
+
+
+class Loop(Expr):
+    body: Expr
+    label: optional[string]
+
+#     Call {
+#         callee: ExprId,
+#         args: Box<[ExprId]>,
+#         is_assignee_expr: bool,
+#     },
+
+
+class Call(Expr):
+    callee: Expr
+    args: list[Expr]
+    is_assignee_expr: predicate
+
+#     MethodCall {
+#         receiver: ExprId,
+#         method_name: Name,
+#         args: Box<[ExprId]>,
+#         generic_args: Option<Box<GenericArgs>>,
+#     },
+
+
+class MethodCall(Expr):
+    receiver: Expr
+    method_name: string
+    args: list[Expr]
+    # TODO
+    # generic_args: optional[GenericArgs]
+
+# pub struct MatchArm {
+#     pub pat: PatId,
+#     pub guard: Option<ExprId>,
+#     pub expr: ExprId,
+# }
+
+
+@qltest.skip
+class MatchArm(Locatable):
+    pat: Pat
+    guard: optional[Expr]
+    expr: Expr
+#     Match {
+#         expr: ExprId,
+#         arms: Box<[MatchArm]>,
+#     },
+
+
+class Match(Expr):
+    expr: Expr
+    branches: list[MatchArm]
+
+#     Continue {
+#         label: Option<LabelId>,
+#     },
+
+
+class Continue(Expr):
+    label: optional[string]
+
+#     Break {
+#         expr: Option<ExprId>,
+#         label: Option<LabelId>,
+#     },
+
+
+class Break(Expr):
+    expr: optional[Expr]
+    label: optional[string]
+
+
+#     Return {
+#         expr: Option<ExprId>,
+#     },
+
+class Return(Expr):
+    expr: optional[Expr]
+#     Become {
+#         expr: ExprId,
+#     },
+
+
+class Become(Expr):
+    expr: Expr
+#     Yield {
+#         expr: Option<ExprId>,
+#     },
+
+
+class Yield(Expr):
+    expr: optional[Expr]
+
+#     Yeet {
+#         expr: Option<ExprId>,
+#     },
+
+
+class Yeet(Expr):
+    expr: optional[Expr]
+#     RecordLit {
+#         path: Option<Box<Path>>,
+#         fields: Box<[RecordLitField]>,
+#         spread: Option<ExprId>,
+#         ellipsis: bool,
+#         is_assignee_expr: bool,
+#     },
+
+
+class RecordLit(Expr):
+    # TODO
+    pass
+
+
+#     Field {
+#         expr: ExprId,
+#         name: Name,
+#     },
+
+class Field(Expr):
+    expr: Expr
+    name: string
+
+#     Await {
+#         expr: ExprId,
+#     },
+
+
+class Await(Expr):
+    expr: Expr
+
+#     Cast {
+#         expr: ExprId,
+#         type_ref: Interned<TypeRef>,
+#     },
+
+
+class Cast(Expr):
+    expr: Expr
+    type_ref: TypeRef
+#     Ref {
+#         expr: ExprId,
+#         rawness: Rawness,
+#         mutability: Mutability,
+#     },
+
+
+class Ref(Expr):
+    expr: Expr
+    is_raw: predicate
+    is_mut: predicate
+#     Box {
+#         expr: ExprId,
+#     },
+
+
+class Box(Expr):
+    expr: Expr
+#     UnaryOp {
+#         expr: ExprId,
+#         op: UnaryOp,
+#     },
+
+
+class UnaryExpr(Expr):
+    expr: Expr
+    op: string
+
+
+#     BinaryOp {
+#         lhs: ExprId,
+#         rhs: ExprId,
+#         op: Option<BinaryOp>,
+#     },
+
+
+class BinaryOp(Expr):
+    lhs: Expr
+    rhs: Expr
+    op: optional[string]
```

Comment:
out of curiosity, do you know what does `op: None` mean?

Reply:
I have no idea, I am equally curious. Just haven't found the time to run an example expression through the pretty_print function.